### PR TITLE
feat: Bump lowest supported k8s version to 1.23

### PR DESCRIPTION
**Pull request by JorTurFer**
Fixes https://github.com/open-telemetry/opentelemetry-operator/issues/2145

### CODE REVIEW

**Comment by JorTurFer** [comment]
There was a typo in the yaml 1.23. I have fixed it and also I have unified the identations with other kind files. Could you trigger tests again please?

**Review by jaronoff97 — APPROVED** [review]
This is fantastic to see, thank you so much.

**Comment by jaronoff97** [comment]
LGTM @pavolloffay mind taking a look and merging?

**Comment by JorTurFer** [comment]
> I think we can get rid of `LEGACY_KIND_VERSION` from GHA

sure, I have removed it, keeping `CURRENT_KIND_VERSION` instead of setting the version directly, but I can get rid of it too if you want

File: internal/manifests/collector/horizontalpodautoscaler.go [hunk 1]
```diff
@@ -60,225 +57,64 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
...
-			utilizationTarget := autoscalingv2beta2.MetricSpec{
-				Type: autoscalingv2beta2.ResourceMetricSourceType,
-				Resource: &autoscalingv2beta2.ResourceMetricSource{
-					Name: corev1.ResourceMemory,
-					Target: autoscalingv2beta2.MetricTarget{
-						Type:               autoscalingv2beta2.UtilizationMetricType,
-						AverageUtilization: otelcol.Spec.Autoscaler.TargetMemoryUtilization,
-					},
+	if otelcol.Spec.Autoscaler.TargetMemoryUtilization != nil {
+		utilizationTarget := autoscalingv2.MetricSpec{
```
Comment (pavolloffay) [inline internal/manifests/collector/horizontalpodautoscaler.go:63]: nit: `memoryTarget`

File: internal/manifests/collector/horizontalpodautoscaler.go [hunk 2]
```diff
@@ -60,225 +57,64 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
...
+			},
 		}
+		metrics = append(metrics, utilizationTarget)
+	}
 
-		targetCPUUtilization := autoscalingv2beta2.MetricSpec{
-			Type: autoscalingv2beta2.ResourceMetricSourceType,
-			Resource: &autoscalingv2beta2.ResourceMetricSource{
+	if otelcol.Spec.Autoscaler.TargetCPUUtilization != nil {
+		targetCPUUtilization := autoscalingv2.MetricSpec{
```
Comment (pavolloffay) [inline internal/manifests/collector/horizontalpodautoscaler.go:77]: nit: `cpuTarget`

**Review by pavolloffay — APPROVED** [review]

**Comment by pavolloffay** [comment]
> sure, I have removed it, keeping CURRENT_KIND_VERSION instead of setting the version directly, but I can get rid of it too if you want

Let's remove it as well

**Comment by pavolloffay** [comment]
CI fialed

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**